packages/react-components/react-tree/src/components/TreeItemLayout/useTreeItemLayoutStyles.styles.ts

```diff
@@ -51,32 +114,80 @@ const useIconStyles = makeStyles({
 const useAsideStyles = makeStyles({
   base: {
     display: 'flex',
-    alignItems: 'center',
     marginLeft: 'auto',
+    ...shorthands.gridArea('aside'),
+    alignItems: 'center',
+    zIndex: 0,
     ...shorthands.padding(0, tokens.spacingHorizontalM),
     ...shorthands.gap(tokens.spacingHorizontalXS),
   },
 });
 
+/**
+ * Styles for the action icon slot
+ */
+const useActionsStyles = makeStyles({
+  base: {
+    display: 'flex',
+    marginLeft: 'auto',
+    ...shorthands.gridArea('aside'),
+    position: 'relative',
```

Comment:
```suggestion
```

I removed this and couldn't see any issues when using another aside component too, did I miss any other use case?

Reply:
This might be some reminiscent of some old implementation